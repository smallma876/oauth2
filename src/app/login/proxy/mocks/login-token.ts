export const loginToken = {
  iss: 'https://accounts.google.com',
  nbf: 1661831589,
  aud: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  sub: '105035299156813441187',
  email: 'sergio.mallma876@gmail.com',
  email_verified: true,
  azp: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  name: 'Sergio Mallma Chavez',
  picture: 'https://lh3.googleusercontent.com/a-/AFdZucq8mJ7sluN0hMjUBWZSj0LDYWfoN4KvV-52JpyI0A=s96-c',
  given_name: 'Sergio',
  family_name: 'Mallma Chavez',
  iat: 1661831889,
  exp: 1661835489,
  jti: '677749ac9724e83aa7155eac897c2135dfcb979d',
};
