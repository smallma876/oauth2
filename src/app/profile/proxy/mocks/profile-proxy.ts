export const mockAlbum = [
  {
    name: 'first',
    url: 'http',
  },
];
