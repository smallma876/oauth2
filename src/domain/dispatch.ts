export interface DispatchObject<T, P = any> {
  type: T;
  payload?: P;
}
