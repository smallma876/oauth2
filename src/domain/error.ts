export interface OauthError {
  message: string;
  status: number;
}
