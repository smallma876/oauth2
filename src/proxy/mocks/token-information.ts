export const TokenInformation = {
  aud: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  azp: '374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com',
  email: 'sergio.mallma876@gmail.com',
  email_verified: true,
  exp: 1662353890,
  family_name: 'Mallma Chavez',
  given_name: 'Sergio',
  iat: 1662350290,
  iss: 'https://accounts.google.com',
  jti: 'ae3f781891bbad12d13c9d411ad5a45e48ccf556',
  name: 'Sergio Mallma Chavez',
  nbf: 1662349990,
  picture: 'https://lh3.googleusercontent.com/a-/AFdZucr57cAutaV8GJ70u1HWwolYmglPELDZSCrfGEFCVQ=s96-c',
  sub: '105035299156813441187',
};
