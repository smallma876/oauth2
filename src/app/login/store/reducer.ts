import produce, { Draft } from 'immer';
import { Dispatch } from 'react';
import { DispatchObject } from '../../../domain/dispatch';

export type DispatchLogin = Dispatch<DispatchObject<LoginActions>>;

export enum LoginActions {
  ClearData = 'CLEAR_DATA',
}

export interface LoginState {}

export const loginState: LoginState = {};

export const loginReducer = produce((draft: Draft<LoginState>, { type, payload }: DispatchObject<LoginActions>) => {
  switch (type) {
    case LoginActions.ClearData:
      break;
    default:
      throw new Error(`Unknown type: ${type}`);
  }
});
