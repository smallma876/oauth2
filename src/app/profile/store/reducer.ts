import produce, { Draft } from 'immer';
import { Dispatch } from 'react';
import { DispatchObject } from '../../../domain/dispatch';

export type DispatchProfile = Dispatch<DispatchObject<ProfileActions>>;

export enum ProfileActions {
  ClearData = 'CLEAR_DATA',
}

export interface ProfileState {}

export const profileState: ProfileState = {};

export const profileReducer = produce(
  (draft: Draft<ProfileState>, { type, payload }: DispatchObject<ProfileActions>) => {
    switch (type) {
      case ProfileActions.ClearData:
        break;
      default:
        throw new Error(`Unknown type: ${type}`);
    }
  },
);
