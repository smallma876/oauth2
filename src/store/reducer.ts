import produce, { Draft } from 'immer';
import { Dispatch } from 'react';
import { LoginToken } from '../app/login/domain/login-token';
import { DispatchObject } from '../domain/dispatch';

export type DispatchApp = Dispatch<DispatchObject<AppActions>>;

export enum AppActions {
  IsFetching = 'IS_FETCHING',
  FinishFetching = 'FINISHED_FETCHING',
  ErrorFetching = 'ERROR_FETCHING',
  SessionExpired = 'SESSION_EXPIRED',
  ShowLoadingBeforeNavigating = 'SHOW_LOADING_BEFORE_NAVIGATING',
  ClearError = 'CLEAR_ERROR',
  ClearData = 'CLEAR_DATA',
  SetUser = 'SET_USER',
  SetToken = 'SET_TOKEN',
}

export interface AppState {
  isFetching: boolean;
  pendingFetches: number;
  sessionExpired: boolean;
  showLoadingBeforeNavigating: boolean;
  user: LoginToken | null;
  token: string;
}

export const initialState: AppState = {
  isFetching: false,
  pendingFetches: 0,
  sessionExpired: false,
  showLoadingBeforeNavigating: false,
  user: null,
  token: '',
};

export const appReducer = produce((draft: Draft<AppState>, { type, payload }: DispatchObject<AppActions>) => {
  switch (type) {
    case AppActions.IsFetching:
      draft.pendingFetches += 1;
      draft.isFetching = true;
      break;
    case AppActions.FinishFetching:
      draft.pendingFetches -= 1;
      draft.isFetching = false;
      break;
    case AppActions.SessionExpired:
      draft.sessionExpired = true;
      break;
    case AppActions.SetUser:
      draft.user = payload as LoginToken;
      break;
    case AppActions.SetToken:
      draft.token = payload as string;
      break;
    default:
      throw new Error(`Unknown type: ${type}`);
  }
});
