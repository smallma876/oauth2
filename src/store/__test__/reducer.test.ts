import { OauthError } from '../../domain/error';
import { TokenInformation } from '../../proxy/mocks/token-information';
import { AppActions, appReducer, initialState } from '../reducer';

describe('reducer test', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should have fetching true', () => {
    const state = { ...initialState, isFetching: true, pendingFetches: 1 };
    const finalState = appReducer(initialState, { type: AppActions.IsFetching });

    expect(finalState).toMatchSnapshot();
    expect(finalState).toEqual(state);
  });

  it('should have fetching false', () => {
    const state = { ...initialState, isFetching: false, pendingFetches: -1 };
    const finalState = appReducer(initialState, { type: AppActions.FinishFetching });

    expect(finalState).toMatchSnapshot();
    expect(finalState).toEqual(state);
  });

  it('should have session expired true', () => {
    const state = { ...initialState, sessionExpired: true };
    const finalState = appReducer(state, { type: AppActions.SessionExpired });

    expect(finalState).toMatchSnapshot();
    expect(finalState).toEqual(state);
  });

  it('should set user', () => {
    const state = { ...initialState, user: TokenInformation };
    const finalState = appReducer(initialState, { type: AppActions.SetUser, payload: TokenInformation });

    expect(finalState).toMatchSnapshot();
    expect(finalState).toEqual(state);
  });

  it('should set token', () => {
    const state = { ...initialState, token: 'XaBCX' };
    const finalState = appReducer(initialState, { type: AppActions.SetToken, payload: 'XaBCX' });

    expect(finalState).toMatchSnapshot();
    expect(finalState).toEqual(state);
  });

  it('should return error', () => {
    try {
      appReducer(initialState, { type: 'test' as AppActions });
    } catch (error) {
      // eslint-disable-next-line jest/no-conditional-expect
      expect((error as OauthError).message).toBe(`Unknown type: test`);
    }
  });
});
