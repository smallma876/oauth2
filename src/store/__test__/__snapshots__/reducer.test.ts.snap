// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer test should have fetching false 1`] = `
Object {
  "isFetching": false,
  "pendingFetches": -1,
  "sessionExpired": false,
  "showLoadingBeforeNavigating": false,
  "token": "",
  "user": null,
}
`;

exports[`reducer test should have fetching true 1`] = `
Object {
  "isFetching": true,
  "pendingFetches": 1,
  "sessionExpired": false,
  "showLoadingBeforeNavigating": false,
  "token": "",
  "user": null,
}
`;

exports[`reducer test should have session expired true 1`] = `
Object {
  "isFetching": false,
  "pendingFetches": 0,
  "sessionExpired": true,
  "showLoadingBeforeNavigating": false,
  "token": "",
  "user": null,
}
`;

exports[`reducer test should set token 1`] = `
Object {
  "isFetching": false,
  "pendingFetches": 0,
  "sessionExpired": false,
  "showLoadingBeforeNavigating": false,
  "token": "XaBCX",
  "user": null,
}
`;

exports[`reducer test should set user 1`] = `
Object {
  "isFetching": false,
  "pendingFetches": 0,
  "sessionExpired": false,
  "showLoadingBeforeNavigating": false,
  "token": "",
  "user": Object {
    "aud": "374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com",
    "azp": "374760273133-4eghe15hb1ek840nmnjotpvtelv2fq11.apps.googleusercontent.com",
    "email": "sergio.mallma876@gmail.com",
    "email_verified": true,
    "exp": 1662353890,
    "family_name": "Mallma Chavez",
    "given_name": "Sergio",
    "iat": 1662350290,
    "iss": "https://accounts.google.com",
    "jti": "ae3f781891bbad12d13c9d411ad5a45e48ccf556",
    "name": "Sergio Mallma Chavez",
    "nbf": 1662349990,
    "picture": "https://lh3.googleusercontent.com/a-/AFdZucr57cAutaV8GJ70u1HWwolYmglPELDZSCrfGEFCVQ=s96-c",
    "sub": "105035299156813441187",
  },
}
`;
